app/src/main/java/com/mdg/droiders/samagra/shush/receivers/AlarmBroadcastReceiver.java

```diff
@@ -0,0 +1,57 @@
+package com.mdg.droiders.samagra.shush.receivers;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.media.AudioManager;
+
+import com.mdg.droiders.samagra.shush.AlarmScheduler;
+import com.mdg.droiders.samagra.shush.R;
+import com.mdg.droiders.samagra.shush.utils.RingerUtils;
+
+import java.util.Calendar;
+
+/**
+ * Created by rohan on 12/8/17.
+ * <br>
+ * Broadcast Receiver that receives a broadcast from the alarm manager to silence/un-silence the phone.
+ */
+public class AlarmBroadcastReceiver extends BroadcastReceiver {
+
+    private static final String LOG_TAG = "Samagra/BR/";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        // The phone will be shushed(silenced) if the boolean shush is true
+        boolean shush = intent.getBooleanExtra(
+                context.getString(R.string.alarm_intent_boolean_extra_key), false);
+        // The timeInMillis at which the alarm is triggered
+        // Used to trigger an alarm for next week at the same time.
+        long timeInMillis = intent.getLongExtra(
+                context.getString(R.string.alarm_intent_long_extra_key), -1);
+        // The alarm id used in pending intent
+        int alarmID = intent.getIntExtra(
+                context.getString(R.string.alarm_intent_int_extra_key), -1);
+
+        if (timeInMillis == -1 || alarmID == -1){
+            return;
```

Comment:
alarmID is SQLite generated primary key and is auto incremented. timeInMillis is a millisecond value that has passed since the epoch. Therefore, neither of them can be negative

Reply:
epoch?

Reply:
See [this](https://developer.android.com/reference/java/util/Calendar.html).
>  An instant in time can be represented by a millisecond value that is an offset from the Epoch, January 1, 1970 00:00:00.000 GMT (Gregorian).